### PR TITLE
feat: avatar #80

### PR DESCRIPTION
**Pull request by mizek1**

### CODE REVIEW

File: projects/ion/src/lib/avatar/avatar.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    });
+  });
+  describe('Icon', () => {
+    it('should render correct icon when type is icon', async () => {
+      await sut({ type: 'icon' });
+      expect(document.getElementById('ion-icon-union')).toBeInTheDocument();
+    });
+  });
+  describe('Photo', () => {
+    it('should render avatar with icon when type is icon', async () => {
```
Comment (lucasrasec) [inline projects/ion/src/lib/avatar/avatar.component.spec.ts:58]:
```suggestion
    it('should render avatar with image when type is photo', async () => {
```

File: projects/ion/src/lib/avatar/avatar.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      await sut({ type: 'photo', image: 'assets/images/avatar.jpg' });
+      expect(screen.getByRole('img')).toHaveAttribute(
+        'src',
+        'assets/images/avatar.jpg'
+      );
+    });
```
Comment (lucasrasec) [inline projects/ion/src/lib/avatar/avatar.component.spec.ts:64]: It shouldn't have a test to cover the on error image?

File: projects/ion/src/lib/avatar/avatar.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    it('should render -- when no value is provided', async () => {
+      await sut({ type: 'initials' });
+      expect(screen.getByText('--')).toBeInTheDocument();
+    });
+    it('should only show initials', async () => {
+      await sut({ type: 'initials', value: 'Taylor Swift' });
+      expect(document.getElementById('ion-icon-union')).not.toBeInTheDocument();
+      expect(screen.queryByRole('img')).not.toBeInTheDocument();
+    });
```
Comment (iurynogueira) [inline projects/ion/src/lib/avatar/avatar.component.spec.ts:49]:
```suggestion
    it('should only show initials', async () => {
      await sut({ type: 'initials', value: 'Taylor Swift' });
      expect(document.getElementById('ion-icon-union')).not.toBeInTheDocument();
      expect(screen.queryByRole('img')).not.toBeInTheDocument();
      expect(screen.getByText('TS')).toBeInTheDocument();
    });
```
If you remove the switchcase of initials this test will pass.

File: projects/ion/src/lib/avatar/avatar.component.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { AvatarType } from '../core/types/avatar';
+import { IconType } from '../icon/icon.component';
+
+export type SizesType = 'lg' | 'md' | 'sm' | 'xs';
```
Comment (iurynogueira) [inline projects/ion/src/lib/avatar/avatar.component.ts:5]: create sizes.ts in core and add this like AvataSizeType

File: projects/ion/src/lib/avatar/avatar.component.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        .split(' ')
+        .map((word) => word[0])
+        .slice(0, 2)
+        .join('')
+        .toUpperCase()
+    );
+  }
+
+  ngOnInit() {
+    if (this.type === 'initials') {
```
Comment (iurynogueira) [inline projects/ion/src/lib/avatar/avatar.component.ts:43]:
```suggestion
    if (this.type === AvatarType.Initials) {
```

particularly I prefer it to be an ENUM as it is easier to maintain

File: projects/ion/src/lib/avatar/avatar.component.spec.ts (hunk 2 above)
Comment (iurynogueira) [reply projects/ion/src/lib/avatar/avatar.component.spec.ts:64]: projects/ion/src/lib/defaultImage.directive.spec.ts test this case.

File: projects/ion/src/lib/defaultImage.directive.spec.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { render, screen } from '@testing-library/angular';
+import { DefaultImageDirective } from './defaultImage.directive';
+
+async function sut(src = 'test.png', defaultImage = 'default.png') {
```
Comment (iurynogueira) [inline projects/ion/src/lib/defaultImage.directive.spec.ts:4]:
```suggestion
const defaultSrc = 'test.png';
async function sut(src = defaultSrc, defaultImage = 'default.png') {
```

Same to defaultImage.

File: projects/ion/src/lib/defaultImage.directive.spec.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  await render(`<img src="${src}" default="${defaultImage}">`, {
+    declarations: [DefaultImageDirective],
+  });
+  return screen.getByRole('img');
+}
+
+describe('Directive: DefaultImage', () => {
+  it('should create an image with directive', async () => {
+    const image = await sut();
+    expect(image).toHaveAttribute('src', 'test.png');
```
Comment (iurynogueira) [inline projects/ion/src/lib/defaultImage.directive.spec.ts:14]:
```suggestion
    expect(image).toHaveAttribute('src', defaultSrc);
```
Same to 'default'

File: projects/ion/src/lib/defaultImage.directive.spec.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    expect(image).toHaveAttribute('default', 'default.png');
+  });
+  it('should set default image when error', async () => {
+    const spy = jest.spyOn(DefaultImageDirective.prototype, 'updateUrl');
+    const image = await sut();
+    image.dispatchEvent(new Event('error'));
+    expect(spy).toBeCalled();
```
Comment (iurynogueira) [inline projects/ion/src/lib/defaultImage.directive.spec.ts:21]: Nice! :D

File: projects/ion/src/lib/icon/svgs/icons.ts [hunk 9]
```diff
@@ -17,4 +17,6 @@ export const iconsPaths: Record<string, string> = {
...
...C2 16.5374 2.46267 17 3.03333 17H5C5.55228 17 6 16.5523 6 16C6 15.4477 5.55228 15 5 15H4.08827C4.42998 13.199 5.73503 12 7.5 12H7.53333C8.00949 12 8.40792 11.6672 8.50871 11.2215C8.51852 11.1949 8.52728 11.1677 8.53494 11.1399C8.68156 10.6074 8.36877 10.0569 7.83631 9.91031C7.2248 9.74192 6.70016 9.34741 6.36866 8.80668C6.03715 8.26595 5.92356 7.61943 6.05091 6.99808C6.17825 6.37673 6.537 5.82702 7.05449 5.46029C7.57198 5.09355 8.20951 4.93722 8.83795 5.02296C9.38517 5.09762 9.88929 4.71454 9.96395 4.16732C10.0386 3.62011 9.65553 3.11598 9.10831 3.04132Z" fill="#282B33"/><path fill-rule="evenodd" clip-rule="evenodd" d="M14 3.00001C11.2386 3.00001 9 5.23859 9 8.00001C9 9.8823 10.0401 11.5217 11.5769 12.3747C8.3594 13.3972 6 16.3983 6 19.9533C6 20.5314 6.46863 21 7.04667 21H20.9533C21.5314 21 22 20.5314 22 19.9533C22 16.3983 19.6406 13.3972 16.4231 12.3747C17.9599 11.5217 19 9.8823 19 8.00001C19 5.23859 16.7614 3.00001 14 3.00001ZM11 8.00001C11 6.34316 12.3431 5.00001 14 5.00001C15.6569 5.00001 17 6.34316 17 8.00001C17 9.65687 15.6569 11 14 11C12.3431 11 11 9.65687 11 8.00001ZM19.9227 19C19.4594 16.1712 16.9706 14 14 14C11.0294 14 8.54061 16.1712 8.07731 19H19.9227Z" fill="#282B33"/>
```
Comment (iurynogueira) [inline projects/ion/src/lib/icon/svgs/icons.ts:20]: Remove fill

File: projects/ion/src/lib/icon/svgs/icons.ts [hunk 10]
```diff
@@ -17,4 +17,6 @@ export const iconsPaths: Record<string, string> = {
...
+  `,
```
Comment (iurynogueira) [inline projects/ion/src/lib/icon/svgs/icons.ts:21]:
```suggestion
  union: `<path d="M9.10831 3.04132C7.97712 2.88699 6.82956 3.16838 5.89808 3.8285C4.9666 4.48862 4.32085 5.47811 4.09163 6.59653C3.86241 7.71496 4.06687 8.87869 4.66358 9.85201C4.80752 10.0868 4.9717 10.3063 5.15342 10.5084C3.15241 11.4567 2 13.6137 2 15.9667C2 16.5374 2.46267 17 3.03333 17H5C5.55228 17 6 16.5523 6 16C6 15.4477 5.55228 15 5 15H4.08827C4.42998 13.199 5.73503 12 7.5 12H7.53333C8.00949 12 8.40792 11.6672 8.50871 11.2215C8.51852 11.1949 8.52728 11.1677 8.53494 11.1399C8.68156 10.6074 8.36877 10.0569 7.83631 9.91031C7.2248 9.74192 6.70016 9.34741 6.36866 8.80668C6.03715 8.26595 5.92356 7.61943 6.05091 6.99808C6.17825 6.37673 6.537 5.82702 7.05449 5.46029C7.57198 5.09355 8.20951 4.93722 8.83795 5.02296C9.38517 5.09762 9.88929 4.71454 9.96395 4.16732C10.0386 3.62011 9.65553 3.11598 9.10831 3.04132Z" fill="#282B33"/><path fill-rule="evenodd" clip-rule="evenodd" d="M14 3.00001C11.2386 3.00001 9 5.23859 9 8.00001C9 9.8823 10.0401 11.5217 11.5769 12.3747C8.3594 13.3972 6 16.3983 6 19.9533C6 20.5314 6.46863 21 7.04667 21H20.9533C21.5314 21 22 20.5314 22 19.9533C22 16.3983 19.6406 13.3972 16.4231 12.3747C17.9599 11.5217 19 9.8823 19 8.00001C19 5.23859 16.7614 3.00001 14 3.00001ZM11 8.00001C11 6.34316 12.3431 5.00001 14 5.00001C15.6569 5.00001 17 6.34316 17 8.00001C17 9.65687 15.6569 11 14 11C12.3431 11 11 9.65687 11 8.00001ZM19.9227 19C19.4594 16.1712 16.9706 14 14 14C11.0294 14 8.54061 16.1712 8.07731 19H19.9227Z"/>`,
```

File: projects/ion/src/public-api.ts [hunk 11]
```diff
@@ -6,6 +6,7 @@ export * from './lib/ion.service';
 export * from './lib/ion.component';
 export * from './lib/ion.module';
 
+export * from './lib/avatar/avatar.component';
```
Comment (iurynogueira) [inline projects/ion/src/public-api.ts:9]: 👍

File: stories/Avatar.stories.ts [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+export const Icon = Template.bind({});
+Icon.args = {
+  type: 'icon',
+};
+
+export const Photo = Template.bind({});
+Photo.args = {
+  type: 'photo',
+  image:
+    'https://64.media.tumblr.com/40e2174ab5e68b1eabbc3dfc78607cef/c1effc67d5c3a1fd-20/s540x810/9d6ce72fcddf97841e7410a0652dd9d5f018b35d.pnj',
```
Comment (iurynogueira) [inline stories/Avatar.stories.ts:36]: Cool img

**Review by iurynogueira — CHANGES_REQUESTED** [review]
Thanks! 👯

**Review by iurynogueira — APPROVED** [review]

**Review by Deeved — APPROVED** [review]

**Pull request merged**